lib/packwerk/node.rb

```diff
@@ -199,8 +200,30 @@ def value_from_hash(hash_node, key)
 
       private
 
+      BLOCK = :block
+      CLASS = :class
+      CONSTANT = :const
+      CONSTANT_ASSIGNMENT = :casgn
+      CONSTANT_ROOT_NAMESPACE = :cbase
+      HASH = :hash
+      HASH_PAIR = :pair
+      METHOD_CALL = :send
+      MODULE = :module
+      SELF = :self
+      STRING = :str
+      SYMBOL = :sym
+
+      private_constant(
+        :BLOCK, :CLASS, :CONSTANT, :CONSTANT_ASSIGNMENT, :CONSTANT_ROOT_NAMESPACE, :HASH, :HASH_PAIR, :METHOD_CALL,
+        :MODULE, :SELF, :STRING, :SYMBOL,
+      )
+
+      def type_of(node)
+        node&.type
```

Comment:
@shageman I brought over this `nil` check from what we had even though you deleted the test for this method on `shopify/packwerk` when it was called `type`. Do we still need this check? Should this be added back to `shopify/packwerk`?